src/main/_docs/setup/setup-openshift.md

```diff
@@ -0,0 +1,99 @@
+curl -fsSL ${DEPLOY_SCRIPT_URL} -o get-che.sh
+curl -fsSL ${WAIT_SCRIPT_URL} -o wait-che.sh
+curl -fsSL ${STACKS_SCRIPT_URL} -o stacks-che.sh
+bash get-che.sh && wait-che.sh && stacks-che.sh
+```
+
+# Deploy Che on Minishift
+
+Download and run deployment scripts
+
+```shell
+DEPLOY_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/deploy_che.sh
+WAIT_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/wait_until_che_is_available.sh
+STACKS_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/replace_stacks.sh
+curl -fsSL ${DEPLOY_SCRIPT_URL} -o get-che.sh
+curl -fsSL ${WAIT_SCRIPT_URL} -o wait-che.sh
+curl -fsSL ${STACKS_SCRIPT_URL} -o stacks-che.sh
+bash get-che.sh && wait-che.sh && stacks-che.sh
+```
+
+# Deploy Che on openshift.io
+
+Use environment variables to set deployment options
+
+```shell
+export OPENSHIFT_TOKEN=<OSO_TOKEN> # Retrieve the OSO_TOKEN from https://console.starter-us-east-2.openshift.com/console/command-line
+export OPENSHIFT_FLAVOR=osio
+```
+
+Download and run deployment scripts
+
+```shell
+DEPLOY_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/deploy_che.sh
+WAIT_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/wait_until_che_is_available.sh
+STACKS_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/replace_stacks.sh
+curl -fsSL ${DEPLOY_SCRIPT_URL} -o get-che.sh
+curl -fsSL ${WAIT_SCRIPT_URL} -o wait-che.sh
+curl -fsSL ${STACKS_SCRIPT_URL} -o stacks-che.sh
+bash get-che.sh && wait-che.sh && stacks-che.sh
+```
+
+# Deployment Options
+
+You can set different deployment options using environment variables:
+
+* `OPENSHIFT_FLAVOR`: possible values are `ocp`, `minishift` and `osio` (default is `minishift`)
+* `OPENSHIFT_ENDPOINT`: url of the OpenShift API (default is unset for ocp, `https://$(minishift ip):8443/` for minishift, `https://api.starter-us-east-2.openshift.com` for osio)
+* `OPENSHIFT_TOKEN` (default is unset)
+* `CHE_OPENSHIFT_PROJECT`: the OpenShift namespace where Che will be deployed (default is `eclipse-che` for ocp and minishift and `${OPENSHIFT_ID}-che` for osio)
+* `CHE_IMAGE_REPO`: `che-server` Docker image repository that will be used for deployment (default is `docker.io/eclipse/che-server`)
+* `CHE_IMAGE_TAG`: `che-server` Docker image tag that will be used for deployment (default is `nightly-centos`)
+* `CHE_LOG_LEVEL`: Log level of che-server (default is `DEBUG`)
+* `CHE_DEBUGGING_ENABLED`: If set to `true` the script will create the OpenShift service to debug che-server (default is `true`)
+* `CHE_KEYCLOAK_DISABLED`: If this is set to true Keycloack authentication will be disabled (default is `true` for ocp and minishift, `false` for osio)
```

Comment:
Not necessary for this release, but shouldn't we be consistent with boolean variables all being `..._ENABLED` versus switching between `ENABLED` and `DISABLED`?

Reply:
I agree. We already had an issue for that 😄 https://github.com/redhat-developer/rh-che/issues/224